language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModuleVariableDeclarationNodeContext.java

```diff
@@ -55,9 +59,12 @@ public ModuleVariableDeclarationNodeContext() {
         List<LSCompletionItem> completionItems = new ArrayList<>();
         ResolvedContext resolvedContext;
         TypeDescriptorNode tDescNode = node.typedBindingPattern().typeDescriptor();
-        if (this.withinInitializerContext(ctx, node)) {
-            completionItems.addAll(this.initializerContextCompletions(ctx, tDescNode));
+        if (node.initializer().isPresent() && this.withinInitializerContext(ctx, node)) {
```

Comment:
Get the sorting invocation out of the if clauses. We can enclose the `return CompletionUtil.route(ctx, node.parent());` within the else clause. Also, send all the routing calls through the overridden `route` method. If we want to call a separate the super's `route` method, do that within the overridden method so the implementation will be consistent